### PR TITLE
IGNITE-8354 Ignite Continuosly refreshes Cassandra Session when there is an Exception in execute method of CassandraSessionImpl

### PR DESCRIPTION
**Pull request by irudyak**

### CODE REVIEW

File: modules/cassandra/store/src/test/bootstrap/aws/cassandra/cassandra-env.sh [hunk 1]
```diff
@@ -130,10 +130,10 @@ esac
 # start-up. For production use you may wish to adjust this for your
 # environment. MAX_HEAP_SIZE is the total amount of memory dedicated
 # to the Java heap. HEAP_NEWSIZE refers to the size of the young
-# generation. Both MAX_HEAP_SIZE and HEAP_NEWSIZE should be either set
+# sesGeneration. Both MAX_HEAP_SIZE and HEAP_NEWSIZE should be either set
```
Comment (sunnychanwork) [inline modules/cassandra/store/src/test/bootstrap/aws/cassandra/cassandra-env.sh]: Why do you substitute generation with sesGeneration here?

File: modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/CassandraSessionImpl.java [hunk 2]
```diff
@@ -50,6 +52,74 @@
  * Implementation for {@link org.apache.ignite.cache.store.cassandra.session.CassandraSession}.
  */
 public class CassandraSessionImpl implements CassandraSession {
+    /**
+     * Simple container for Cassandra session and its generation number.
+     */
+    private static class WrappedSession {
```
Comment (sunnychanwork) [inline modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/CassandraSessionImpl.java]: Considering that you have moved WrapperPrepareStatement in a seperate file, would you want to do the same for WrappedSession?

File: modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/CassandraSessionImpl.java [hunk 3]
```diff
@@ -832,77 +954,53 @@ private void handleTableAbsenceError(String table, KeyValuePersistenceSettings s
...
+        }
 
-        try {
-            synchronized (prepStatementHandlersCnt) {
-                // Oooops... I am not the first thread who tried to handle prepared statement problem.
-                if (hndNum != 0) {
-                    log.warning("Prepared statement cluster error detected, another thread already fixed the problem", e);
-                    return;
-                }
+        synchronized (generation) {
```
[the remark this answers is not in the document]
Comment (irudyak) [reply modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/CassandraSessionImpl.java]: Thanks makes sense

File: modules/cassandra/store/src/test/bootstrap/aws/cassandra/cassandra-env.sh (hunk 1 above)
Comment (irudyak) [reply modules/cassandra/store/src/test/bootstrap/aws/cassandra/cassandra-env.sh]: Oh, that's just a weird behavior of IDE refactoring, going to revert it

File: modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/CassandraSessionImpl.java (hunk 2 above)
Comment (irudyak) [reply modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/CassandraSessionImpl.java]: Yeah, actually it'll make code more clean

File: modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/WrappedPreparedStatement.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.PreparedId;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.policies.RetryPolicy;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ * Simple wrapper providing access to Cassandra prepared statement and sesGeneration of Cassandra
```
Comment (akuznetsov-gridgain) [inline modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/WrappedPreparedStatement.java]: sesGeneration?

File: modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/WrappedPreparedStatement.java [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+ * session which was used to create this statement
+ */
+public class WrappedPreparedStatement implements PreparedStatement {
+    /** Prepared statement. **/
+    public final PreparedStatement st;
```
Comment (akuznetsov-gridgain) [inline modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/WrappedPreparedStatement.java]: Do we really need this fields public?

File: modules/cassandra/store/src/test/bootstrap/aws/cassandra/cassandra-env.sh [hunk 6]
```diff
@@ -130,10 +130,10 @@ esac
...
 # or not (if you set one, set the other).
 #
-# The main trade-off for the young generation is that the larger it
+# The main trade-off for the young sesGeneration is that the larger it
```
Comment (akuznetsov-gridgain) [inline modules/cassandra/store/src/test/bootstrap/aws/cassandra/cassandra-env.sh]: sesGeneration ?

File: modules/cassandra/store/src/test/java/org/apache/ignite/tests/CassandraConfigTest.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
+package org.apache.ignite.tests;
+
+import junit.framework.TestCase;
+import org.apache.ignite.cache.affinity.AffinityKeyMapped;
+import org.apache.ignite.cache.query.annotations.QuerySqlField;
+import org.apache.ignite.cache.store.cassandra.persistence.KeyPersistenceSettings;
+import org.apache.ignite.cache.store.cassandra.persistence.KeyValuePersistenceSettings;
+
+public class CassandraConfigTest extends TestCase {
```
Comment (akuznetsov-gridgain) [inline modules/cassandra/store/src/test/java/org/apache/ignite/tests/CassandraConfigTest.java]: Missing javadoc on class

File: modules/cassandra/store/src/test/java/org/apache/ignite/tests/CassandraConfigTest.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
+package org.apache.ignite.tests;
```
Comment (akuznetsov-gridgain) [inline modules/cassandra/store/src/test/java/org/apache/ignite/tests/CassandraConfigTest.java]: Missing license header.

File: modules/cassandra/store/src/test/java/org/apache/ignite/tests/CassandraConfigTest.java (hunk 7 above)
Comment (irudyak) [reply modules/cassandra/store/src/test/java/org/apache/ignite/tests/CassandraConfigTest.java]: Makes sense

File: modules/cassandra/store/src/test/java/org/apache/ignite/tests/CassandraConfigTest.java (hunk 8 above)
Comment (irudyak) [reply modules/cassandra/store/src/test/java/org/apache/ignite/tests/CassandraConfigTest.java]: Makes sense

File: modules/cassandra/store/src/test/bootstrap/aws/cassandra/cassandra-env.sh (hunk 6 above)
Comment (irudyak) [reply modules/cassandra/store/src/test/bootstrap/aws/cassandra/cassandra-env.sh]: oh... thanks for finding this, that's a result of using IDE refactoring tool

File: modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/WrappedPreparedStatement.java (hunk 4 above)
Comment (irudyak) [reply modules/cassandra/store/src/main/java/org/apache/ignite/cache/store/cassandra/session/WrappedPreparedStatement.java]: Thanks, that's weird result of refactoring

**Comment by irudyak** [comment]
All the comments were addressed

**Comment by sunnychanwork** [comment]
Looks much better - thank you.

**Comment by irudyak** [comment]
Changes were merged into Apache master branch

**Pull request closed without merging**